aten/src/ATen/native/Convolution.cpp

```diff
@@ -618,6 +618,9 @@ at::Tensor _convolution(
     bool benchmark, bool deterministic, bool cudnn_enabled) {
 
   const bool input_is_mkldnn = input_r.is_mkldnn();
+  if (input_is_mkldnn) {
```

Comment:
Is it enough to check only for mkldnn inputs? Currently the following (which, as far as I understand, sends regular tensor to mkldnn convolution) works:
```
x = torch.randn(N, C, 56, 56, dtype=torch.float)#.to_mkldnn()
conv2d = torch.nn.Conv2d(in_channels=C,
                            out_channels=M,
                            kernel_size=3,
                            padding=1)
mkldnn_conv2d = mkldnn_utils.to_mkldnn(copy.deepcopy(conv2d))
with torch.backends.mkldnn.flags(enabled=True):
    y_mkldnn = mkldnn_conv2d(x)#.to_dense()
```
Same question for all other functions - you added checks only for mkldnn inputs.

Reply:
@ngimel , for convolution, this input check can be removed, see
1. torch.backends.mkldnn.flags(enabled=True), input and weight are MKLNDN tensor, it is ok.
2. torch.backends.mkldnn.flags(enabled=True), input is dense tensor, and weight are MKLNDN tensor, will call MKLDNN path, there will make check to make sure input and weight have same data type: see 
https://github.com/pytorch/pytorch/blob/6468bc46375e4cb73fe3b59c7b81dff559bb0967/aten/src/ATen/native/Convolution.cpp#L746-L748, so this will report a error: **Input type (torch.FloatTensor) and weight type (Mkldnntorch.FloatTensor) should be the same**
3. torch.backends.mkldnn.flags(enabled=False), input and weight are MKLNDN tensor,  will call this path 
https://github.com/pytorch/pytorch/blob/6468bc46375e4cb73fe3b59c7b81dff559bb0967/aten/src/ATen/native/Convolution.cpp#L801-L802, which not support MKLDNN inputs. So this will report a error: **"opaque tensors do not have is_contiguous"**.
4.  torch.backends.mkldnn.flags(enabled=False), input is dense tensor,  weight are MKLNDN tensor, also call path 
https://github.com/pytorch/pytorch/blob/6468bc46375e4cb73fe3b59c7b81dff559bb0967/aten/src/ATen/native/Convolution.cpp#L801-L802
this will report a error: **Cannot access data pointer of Tensor that doesn't have storage**